scripts/services/emoncms_mqtt/emoncms_mqtt.php

```diff
@@ -334,7 +334,19 @@ function message($message)
                         $input_name = implode("_",$input_name_parts)."_";
                     }
                     foreach ($jsondata as $key=>$value) {
-                        $inputs[] = array("userid"=>$userid, "time"=>$time, "nodeid"=>$nodeid, "name"=>$input_name.$key, "value"=>$value);
+                        // Unbox { name: xxx, value: xxx }
+                        if (is_array($value) && array_key_exists("value", $value)) {
+                            if (array_key_exists("name", $value) && $value["name"])
+                                $key .= '_'.$value["name"];
+                            $value = $value["value"];
+                        }
+
+                        if (is_scalar($value)) {
```

Comment:
I quite like this check, but might I suggest it should check for a number instead?

Reply:
Yes true and yes sort of dropped later, but often create bad inputs as a result so ensuring only numbers are sent on would be a really good thing (but could be proposed as a separate PR).

Reply:
Would be nice to know if there's a possible approach for keeping the strings in the pipeline, if we could add a feed function to do matching or mapping somehow...?